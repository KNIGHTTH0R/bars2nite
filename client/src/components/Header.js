import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import * as actions from '../store/actions';
import SideNav from './SideNav';

export class Header extends Component {
  state = {
    searchInput: '',
    sideNavShow: false
  };

  inputChangeHandler = e => {
    this.setState({ searchInput: e.target.value });
  };

  keyPressHandler = e => {
    if (e.keyCode === 13) {
      e.preventDefault();
      this.onSearchBars();
    }
  };

  toggleSideNav = () => {
    this.setState(prevState => {
      return { sideNavShow: !prevState.sideNavShow };
    });
  };

  onSearchBars = () => {
    if (this.state.searchInput !== '') {
      this.props.onSearchBars(this.state.searchInput);
      this.props.history.push('/search');
      this.setState({ searchInput: '' });
    }
  };

  render() {
    return (
      <div className="Header">
        <nav className="navigation">
          <span className="navigation__header">
            <Link to="/">
              <i className="fa fa-glass glass-icon--1" aria-hidden="true" />
              <i className="fa fa-glass glass-icon--2" aria-hidden="true" />
            </Link>
          </span>
          <form action="#" className="search">
            <input
              type="text"
              className="search__input"
              placeholder="Your city or zip code"
              value={this.state.searchInput}
              onChange={this.inputChangeHandler}
              onKeyDown={this.keyPressHandler}
            />
            <i
              className="fa fa-search search__icon"
              aria-hidden="true"
              onClick={this.onSearchBars}
            />
          </form>

          <ul className="navigation__list">
            <li className="sidenav-icon">
              <i
                onClick={this.toggleSideNav}
                className="fa fa-bars"
                aria-hidden="true"
              />
            </li>

            <li
              className="SideNav"
              style={{
                marginRight: this.state.sideNavShow ? '0' : '-150px',
                opacity: this.state.sideNavShow ? 1 : 0
              }}
            >
              <SideNav />
            </li>
          </ul>
        </nav>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    auth: state.auth
  };
}

export default connect(
  mapStateToProps,
  actions
)(withRouter(Header));
