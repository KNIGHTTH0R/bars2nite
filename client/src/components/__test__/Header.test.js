import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../Header';

describe('Header', () => {
  let header = shallow(<Header />);

  it('renders correctly', () => {
    expect(header).toMatchSnapshot();
  });

  describe('when user types in the search input', () => {
    const search_input = 'los angeles';

    it('updates the `searchInput` in the state', () => {
      header
        .find('.search__input')
        .simulate('change', { target: { value: search_input } });
      expect(header.state().searchInput).toEqual(search_input);
    });
  });

  describe('when user clicks the search button', () => {
    const mockOnSearchBar = jest.fn();
    const props = {
      onSearchBars: mockOnSearchBar,
      history: []
    };
    header = shallow(<Header {...props} />);

    beforeEach(() => {
      header.find('.fa.fa-search.search__icon').simulate('click');
    });

    it('clears the `searchInput` in the state', () => {
      expect(header.state().searchInput).toEqual('');
    });

    it('dispatches the `onSearchBar` it receives in props', () => {
      expect(mockOnSearchBar).toHaveBeenCalled();
    });
  });

  describe('when user clicks the nav button', () => {
    it('updates the `sideNavShow` in the state', () => {
      header.find('.navigation__list').simulate('click');

      expect(header.state().sideNavShow).toBe(true);
    });
  });
});
