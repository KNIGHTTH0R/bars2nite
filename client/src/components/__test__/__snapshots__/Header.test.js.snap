// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    history={Array []}
    onSearchBars={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <nav
        className="navigation"
>
        <span
                className="navigation__header"
        >
                <Link
                        replace={false}
                        to="/"
                >
                        <i
                                aria-hidden="true"
                                className="fa fa-glass glass-icon--1"
                        />
                        <i
                                aria-hidden="true"
                                className="fa fa-glass glass-icon--2"
                        />
                </Link>
        </span>
        <form
                action="#"
                className="search"
        >
                <input
                        className="search__input"
                        onChange={[Function]}
                        onKeyDown={[Function]}
                        placeholder="Your city or zip code"
                        type="text"
                        value=""
                />
                <i
                        aria-hidden="true"
                        className="fa fa-search search__icon"
                        onClick={[Function]}
                />
        </form>
        <ul
                className="navigation__list"
                onClick={[Function]}
        >
                <li
                        className="sidenav-icon"
                >
                        <span
                                className="sidenav-icon--line"
                        />
                </li>
                <li
                        className="SideNav"
                        style={
                                Object {
                                        "marginRight": "-150px",
                                        "opacity": 0,
                                      }
                        }
                >
                        <Connect(SideNav) />
                </li>
        </ul>
</nav>,
      "className": "Header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="navigation__header"
>
            <Link
                        replace={false}
                        to="/"
            >
                        <i
                                    aria-hidden="true"
                                    className="fa fa-glass glass-icon--1"
                        />
                        <i
                                    aria-hidden="true"
                                    className="fa fa-glass glass-icon--2"
                        />
            </Link>
</span>,
          <form
            action="#"
            className="search"
>
            <input
                        className="search__input"
                        onChange={[Function]}
                        onKeyDown={[Function]}
                        placeholder="Your city or zip code"
                        type="text"
                        value=""
            />
            <i
                        aria-hidden="true"
                        className="fa fa-search search__icon"
                        onClick={[Function]}
            />
</form>,
          <ul
            className="navigation__list"
            onClick={[Function]}
>
            <li
                        className="sidenav-icon"
            >
                        <span
                                    className="sidenav-icon--line"
                        />
            </li>
            <li
                        className="SideNav"
                        style={
                                    Object {
                                                "marginRight": "-150px",
                                                "opacity": 0,
                                              }
                        }
            >
                        <Connect(SideNav) />
            </li>
</ul>,
        ],
        "className": "navigation",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/"
>
              <i
                            aria-hidden="true"
                            className="fa fa-glass glass-icon--1"
              />
              <i
                            aria-hidden="true"
                            className="fa fa-glass glass-icon--2"
              />
</Link>,
            "className": "navigation__header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <i
                  aria-hidden="true"
                  className="fa fa-glass glass-icon--1"
/>,
                <i
                  aria-hidden="true"
                  className="fa fa-glass glass-icon--2"
/>,
              ],
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-hidden": "true",
                  "className": "fa fa-glass glass-icon--1",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-hidden": "true",
                  "className": "fa fa-glass glass-icon--2",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
            ],
            "type": [Function],
          },
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "action": "#",
            "children": Array [
              <input
                className="search__input"
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="Your city or zip code"
                type="text"
                value=""
/>,
              <i
                aria-hidden="true"
                className="fa fa-search search__icon"
                onClick={[Function]}
/>,
            ],
            "className": "search",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "search__input",
                "onChange": [Function],
                "onKeyDown": [Function],
                "placeholder": "Your city or zip code",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-hidden": "true",
                "className": "fa fa-search search__icon",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="sidenav-icon"
>
                <span
                                className="sidenav-icon--line"
                />
</li>,
              <li
                className="SideNav"
                style={
                                Object {
                                                "marginRight": "-150px",
                                                "opacity": 0,
                                              }
                }
>
                <Connect(SideNav) />
</li>,
            ],
            "className": "navigation__list",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span
                  className="sidenav-icon--line"
/>,
                "className": "sidenav-icon",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "sidenav-icon--line",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(SideNav) />,
                "className": "SideNav",
                "style": Object {
                  "marginRight": "-150px",
                  "opacity": 0,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "nav",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav
          className="navigation"
>
          <span
                    className="navigation__header"
          >
                    <Link
                              replace={false}
                              to="/"
                    >
                              <i
                                        aria-hidden="true"
                                        className="fa fa-glass glass-icon--1"
                              />
                              <i
                                        aria-hidden="true"
                                        className="fa fa-glass glass-icon--2"
                              />
                    </Link>
          </span>
          <form
                    action="#"
                    className="search"
          >
                    <input
                              className="search__input"
                              onChange={[Function]}
                              onKeyDown={[Function]}
                              placeholder="Your city or zip code"
                              type="text"
                              value=""
                    />
                    <i
                              aria-hidden="true"
                              className="fa fa-search search__icon"
                              onClick={[Function]}
                    />
          </form>
          <ul
                    className="navigation__list"
                    onClick={[Function]}
          >
                    <li
                              className="sidenav-icon"
                    >
                              <span
                                        className="sidenav-icon--line"
                              />
                    </li>
                    <li
                              className="SideNav"
                              style={
                                        Object {
                                                  "marginRight": "-150px",
                                                  "opacity": 0,
                                                }
                              }
                    >
                              <Connect(SideNav) />
                    </li>
          </ul>
</nav>,
        "className": "Header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="navigation__header"
>
              <Link
                            replace={false}
                            to="/"
              >
                            <i
                                          aria-hidden="true"
                                          className="fa fa-glass glass-icon--1"
                            />
                            <i
                                          aria-hidden="true"
                                          className="fa fa-glass glass-icon--2"
                            />
              </Link>
</span>,
            <form
              action="#"
              className="search"
>
              <input
                            className="search__input"
                            onChange={[Function]}
                            onKeyDown={[Function]}
                            placeholder="Your city or zip code"
                            type="text"
                            value=""
              />
              <i
                            aria-hidden="true"
                            className="fa fa-search search__icon"
                            onClick={[Function]}
              />
</form>,
            <ul
              className="navigation__list"
              onClick={[Function]}
>
              <li
                            className="sidenav-icon"
              >
                            <span
                                          className="sidenav-icon--line"
                            />
              </li>
              <li
                            className="SideNav"
                            style={
                                          Object {
                                                        "marginRight": "-150px",
                                                        "opacity": 0,
                                                      }
                            }
              >
                            <Connect(SideNav) />
              </li>
</ul>,
          ],
          "className": "navigation",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/"
>
                <i
                                aria-hidden="true"
                                className="fa fa-glass glass-icon--1"
                />
                <i
                                aria-hidden="true"
                                className="fa fa-glass glass-icon--2"
                />
</Link>,
              "className": "navigation__header",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <i
                    aria-hidden="true"
                    className="fa fa-glass glass-icon--1"
/>,
                  <i
                    aria-hidden="true"
                    className="fa fa-glass glass-icon--2"
/>,
                ],
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "className": "fa fa-glass glass-icon--1",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "className": "fa fa-glass glass-icon--2",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
              ],
              "type": [Function],
            },
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "action": "#",
              "children": Array [
                <input
                  className="search__input"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Your city or zip code"
                  type="text"
                  value=""
/>,
                <i
                  aria-hidden="true"
                  className="fa fa-search search__icon"
                  onClick={[Function]}
/>,
              ],
              "className": "search",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "search__input",
                  "onChange": [Function],
                  "onKeyDown": [Function],
                  "placeholder": "Your city or zip code",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-hidden": "true",
                  "className": "fa fa-search search__icon",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
            ],
            "type": "form",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="sidenav-icon"
>
                  <span
                                    className="sidenav-icon--line"
                  />
</li>,
                <li
                  className="SideNav"
                  style={
                                    Object {
                                                      "marginRight": "-150px",
                                                      "opacity": 0,
                                                    }
                  }
>
                  <Connect(SideNav) />
</li>,
              ],
              "className": "navigation__list",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span
                    className="sidenav-icon--line"
/>,
                  "className": "sidenav-icon",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "sidenav-icon--line",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Connect(SideNav) />,
                  "className": "SideNav",
                  "style": Object {
                    "marginRight": "-150px",
                    "opacity": 0,
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "nav",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
